sdk/core/Azure.Core/tests/HttpPipelineFunctionalTests.cs

```diff
@@ -467,6 +468,71 @@ public async Task RetriesBufferedBodyTimeout()
             testDoneTcs.Cancel();
         }
 
+        [Test]
+        public void TimeoutsResponseBuffering()
+        {
+            var testDoneTcs = new CancellationTokenSource();
+            HttpPipeline httpPipeline = HttpPipelineBuilder.Build(new TestOptions
+            {
+                Retry =
+                {
+                    NetworkTimeout = TimeSpan.FromMilliseconds(500),
+                    MaxRetries = 0
+                }
+            });
+
+            using TestServer testServer = new TestServer(
+                async _ =>
+                {
+                    await Task.Delay(Timeout.Infinite, testDoneTcs.Token);
+                });
+
+            using HttpMessage message = httpPipeline.CreateMessage();
+            message.Request.Uri.Reset(testServer.Address);
+            message.BufferResponse = true;
+
+            var exception = Assert.ThrowsAsync<TaskCanceledException>(async () => await ExecuteRequest(message, httpPipeline));
+            Assert.AreEqual("The operation was cancelled because it exceeded the configured timeout of 0:00:00.5. " +
```

Comment:
I worry this may not always be 5s. We've had various problems with timeouts across KV and Search, at least. Maybe a `StringAssert.StartsWith` or something? Or maybe just keep that in your back pocket should an issue arise.

Reply:
We set the timeout as part of the test.

Reply:
I know, but the smaller the unit I worry that split-second from a somewhat imprecise clock may not always correlate exactly to the actual timeout. Or...maybe I'm looking at this wrong. This is the value that's formatted, right? Ugh, never mind if that's the case.

Reply:
Yeah, we don't measure we just use the value from options.